website/docs/topics/long_contexts.md

```diff
@@ -0,0 +1,305 @@
+)
+# Define your llm config
+llm_config = {"config_list": config_list}
+```
+
+```{=mdx}
+:::tip
+Learn more about configuring LLMs for agents [here](/docs/topics/llm_configuration).
+:::
+```
+
+```python
+# Define your agent; the user proxy and an assistant
+assistant = autogen.AssistantAgent(
+    "assistant",
+    llm_config=llm_config,
+)
+user_proxy = autogen.UserProxyAgent(
+    "user_proxy",
+    human_input_mode="NEVER",
+    is_termination_msg=lambda x: "TERMINATE" in x.get("content", ""),
+    max_consecutive_auto_reply=10,
+)
+```
+
+### Example 1: Limiting the Total Number of Messages
+
+Consider a scenario where you want to limit the context history to only the most recent messages to maintain efficiency and relevance. You can achieve this with the MessageHistoryLimiter transformation:
+
+```python
+# Limit the message history to the 3 most recent messages
+max_msg_transfrom = transforms.MessageHistoryLimiter(max_messages=3)
+
+messages = [
+    {"role": "user", "content": "hello"},
+    {"role": "assistant", "content": [{"type": "text", "text": "there"}]},
+    {"role": "user", "content": "how"},
+    {"role": "assistant", "content": [{"type": "text", "text": "are you doing?"}]},
+    {"role": "user", "content": "very very very very very very long string"},
+]
+
+processed_messages = max_msg_transfrom.apply_transform(copy.deepcopy(messages))
+pprint.pprint(processed_messages)
+```
+
+```console
+[{'content': 'how', 'role': 'user'},
+{'content': [{'text': 'are you doing?', 'type': 'text'}], 'role': 'assistant'},
+{'content': 'very very very very very very long string', 'role': 'user'}]
+```
+
+By applying the `MessageHistoryLimiter`, we can see that we limited the context history to the 3 most recent messages.
+
+### Example 2: Limiting the Number of Tokens
+
+To adhere to token limitations, use the `MessageTokenLimiter` transformation. This limits tokens per message and the total token count across all messages:
+
+```python
+# Limit the token limit per message to 3 tokens
+token_limit_transform = transforms.MessageTokenLimiter(max_tokens_per_message=3)
+
+processed_messages = token_limit_transform.apply_transform(copy.deepcopy(messages))
+
+pprint.pprint(processed_messages)
+```
+
+```console
+[{'content': 'hello', 'role': 'user'},
+{'content': [{'text': 'there', 'type': 'text'}], 'role': 'assistant'},
+{'content': 'how', 'role': 'user'},
+{'content': [{'text': 'are you doing', 'type': 'text'}], 'role': 'assistant'},
+{'content': 'very very very', 'role': 'user'}]
+```
+
+We can see that we can limit the number of tokens to 3, which is equivalent to 3 words in this instance.
+
+### Example 3: Combining Multiple Transformations Using the `TransformMessages` Capability
+
+Let's test these transforms with AutoGen's agents. We will see that the agent without the capability to handle long context will result in an error, while the agent with the capability will have no issues.
+
+```python
+llm_config = {
```

Comment:
We should not be sharing the `llm_config` across agents. It leads to problem as registering functions will update all agents llm_config.